org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/PropertyUtil.xtend

```diff
@@ -50,21 +51,26 @@ class PropertyUtil {
 	}
 
 	/**
-	 * Returns the name as a property name, e.g. a prefix {@code get}, {@code is} or {@code set}
-	 * can be used with the result of this method.
+	 * Returns the shorthand name for a function
+	 * e.g. URI for getURI; uri for setUri; uri for getuRI; enum for isEnum
 	 * If the given name is invalid, the result is <code>null</code>.
 	 * 
+	 * @param fullName the fullName of the function we're trying to shorthand (including 'get' or 'is')
+	 * 
 	 * @since 2.15
 	 */
 	/* @Nullable */
-	def static String tryGetAsPropertyName(String name) {
-		if (name.length() == 1) { // e.g. Point.getX()
-			if (Character.isUpperCase(name.charAt(0))) {
-				// X is not a valid sugar for getX()
-				return null;
+	def static String tryGetShorthandName(String fullName) {
```

Comment:
@szarnekow I changed the name here to avoid confusion with the previous tryGetAsPropertyName. Is this ok?

Reply:
We haven't used shorthand before in that context so I'm not 100% convinced.

Reply:
@szarnekow Then what would be the correct terminology?

Reply:
Having looked at the JavaBeans [spec](http://download.oracle.com/otndocs/jcp/7224-javabeans-1.01-fr-spec-oth-JSpec/) in the meantime, I believe the correct name would be `inferred property name` (see 8.8 in the spec).

Reply:
I like `inferred property name`!